qa/integration-tests/src/test/java/io/camunda/it/client/SearchDecisionRequirementsTest.java

```diff
@@ -0,0 +1,306 @@
+    // when
+    final var resultAsc =
+        zeebeClient.newDecisionRequirementsQuery().sort(s -> s.version().asc()).send().join();
+    final var resultDesc =
+        zeebeClient.newDecisionRequirementsQuery().sort(s -> s.version().desc()).send().join();
+
+    // Extract unique names from the results
+    final List<Integer> uniqueAscVersions =
+        resultAsc.items().stream()
+            .map(item -> item.getVersion())
+            .distinct()
+            .collect(Collectors.toList());
+
+    final List<Integer> uniqueDescVersions =
+        resultDesc.items().stream()
+            .map(item -> item.getVersion())
+            .distinct()
+            .collect(Collectors.toList());
+
+    // Ensure there are at least two unique names to compare
+    assertThat(uniqueAscVersions.size()).isGreaterThan(1);
+    assertThat(uniqueDescVersions.size()).isGreaterThan(1);
+
+    // Assert that the first unique name in ascending order is less than the second unique name
+    assertThat(uniqueAscVersions.get(0)).isLessThan(uniqueAscVersions.get(1));
+
+    // Assert that the first unique name in descending order is greater than the second unique name
+    assertThat(uniqueDescVersions.get(0)).isGreaterThan(uniqueDescVersions.get(1));
+  }
+
+  @Test
+  public void shouldValidatePagination() {
+    final var result =
+        zeebeClient.newDecisionRequirementsQuery().page(p -> p.limit(1)).send().join();
+    assertThat(result.items().size()).isEqualTo(1);
+    final var key = result.items().getFirst().getDecisionRequirementsKey();
+    // apply searchAfter
+    final var resultAfter =
+        zeebeClient
+            .newDecisionRequirementsQuery()
+            .page(p -> p.searchAfter(Collections.singletonList(key)))
+            .send()
+            .join();
+
+    assertThat(resultAfter.items().size()).isEqualTo(2);
+    final var keyAfter = resultAfter.items().getFirst().getDecisionRequirementsKey();
+    // apply searchBefore
+    final var resultBefore =
+        zeebeClient
+            .newDecisionRequirementsQuery()
+            .page(p -> p.searchBefore(Collections.singletonList(keyAfter)))
+            .send()
+            .join();
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(resultBefore.items().getFirst().getDecisionRequirementsKey()).isEqualTo(key);
+  }
+
+  private static void waitForDecisionRequirementsBeingExported() {
+    Awaitility.await("should receive data from ES")
+        .atMost(Duration.ofMinutes(1))
+        .ignoreExceptions() // Ignore exceptions and continue retrying
+        .untilAsserted(
+            () -> {
+              final var result = zeebeClient.newDecisionRequirementsQuery().send().join();
+              assertThat(result.items().size()).isEqualTo(3);
+
+              // Store the decision requirements key for later use for filter tests
+              decisionRequirementKey = result.items().get(0).getDecisionRequirementsKey();
```

Comment:
you can fetch these data from `zeebeClient.newDeployResourceCommand().addResourceFromClasspath(resourceName).send().join()`

It returns a `DeploymentEvent` which contains all the data that you need
```java
public interface DeploymentEvent {
  /**
   * @return the unique key of the deployment
   */
  long getKey();

  /**
   * @return the processes which are deployed
   */
  List<Process> getProcesses();

  /**
   * @return the decisions which are deployed
   */
  List<Decision> getDecisions();

  /**
   * @return the decision requirements which are deployed
   */
  List<DecisionRequirements> getDecisionRequirements();
```

Reply:
This is true, just a different approach 
As the complexity is related to a variable assignment only, I would prefer keep  as is. 
On this case there are different forms in fact